src/tictacclojure/console_ui.clj

```diff
@@ -0,0 +1,28 @@
+(ns tictacclojure.console-ui)
+
+(defn print-out
+  [message]
+  (println message))
+
+(defn- always-validate
+  [user-input]
+  true)
+
+(defn get-input
+  ([]
+    (get-input "" always-validate ""))
+
+  ([prompt]
+    (get-input prompt always-validate ""))
+
+  ([validate error-message]
+   (get-input "" validate error-message))
+
+  ([prompt validate error-message]
+    (if (not-empty prompt) (print-out prompt))
+    (let [user-input (read-line)]
+      (if (validate user-input)
```

Comment:
mm from the community style guide function inputs should be `f`, `g`, `h`... but I am not sure I like that as it does not say what the function is-does..perhaps a convention of prefixing any such inputs with `f-` might be good? like in this case it would be `f-is-valid?`